### PR TITLE
ref: Add typealias for bytes

### PR DESCRIPTION
**Pull request by kevinrenskers**
This should make it clear what kind of size these variables are. Are they KB, MB, GB? Nope, they are all bytes.

#skip-changelog

### CODE REVIEW

File: Sources/Sentry/include/SentryCrashWrapper.h [hunk 1]
```diff
@@ -3,6 +3,8 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+typedef uint64_t bytes;
```
Comment (philipphofmann) [inline Sources/Sentry/include/SentryCrashWrapper.h:6]: `l`: What about adding a file called `SentryInternalDefines.h` or something to avoid duplication?

**Review by philipphofmann — APPROVED** [review]
Thanks

**Review by philipphofmann — APPROVED** [review]
LGTM

**Review by armcknight — COMMENTED** [review]
There are still usages of `uint64_t` in SentryCrashMonitor_System.m that would intermix with this typedef, this only got about half of them. Did you intend to not change those?

Nice change overall. Relatedly, one of the only times I like using a sort of Hungarian notation is to put units like `_bytes`/`_mb`/`_ms`/`_minutes`/`_degrees`/`_radians` on variable/function names. Also relatedly, one of my favorite reasons to use Swift is to be able to use https://github.com/pointfreeco/swift-tagged for even better type safety around this kind of thing.

**Comment by kevinrenskers** [comment]
I must've missed those, I'll take a look tomorrow. And yea I'm a huge fan of everything the Point-Free guys do :)

**Comment by brustolin** [comment]
TBH Im not a huge fan of this change. The result information should be clear in the documentation, and the function name should be clear.

Having `bytes` as a value type is confusing, nothing indicates that this is a number indicating the amount of bytes. This could easily be misunderstood as a collection of byte (which btw is a valid type in other languages).

Take `- (bytes)freeMemory` for example. Am I freeing some memory? Is this function returning the actual bytes released as a collection?   What about  `- (bytes)freeStorage` what storage in being released? 
I would rather call `freeMemorySize`, `freeStorageSize` and 'appMemorySize'

**Comment by kevinrenskers** [comment]
We can do both. If we call it `freeStorageSize` then question is still "what is the unit", which has come up in multiple comments already.

**Comment by brustolin** [comment]
IMO a simple
```
/**
* Returns the amount of bytes...
*/
```
Above the function in the header should be enough to solve this.

That being said, I will not oppose if you want to merge this. Just sharing my thoughts.

File: Sources/Sentry/include/SentryCrashWrapper.h [hunk 2]
```diff
@@ -1,8 +1,11 @@
 #import "SentryDefines.h"
+#import "SentryInternalDefines.h"
 #import <Foundation/Foundation.h>
 
 NS_ASSUME_NONNULL_BEGIN
 
+typedef uint64_t bytes;
```
Comment (brustolin) [inline Sources/Sentry/include/SentryCrashWrapper.h:7]: since we have `SentryInternalDefines.h` now, do we still need this?

Comment (kevinrenskers) [reply Sources/Sentry/include/SentryCrashWrapper.h]: Nope, good point

**Comment by armcknight** [comment]
We could always add the `_bytes` suffix to those methods to show that is the unit that is being returned. Alternatively, we could define a struct like 
```
NS_ENUM(NSUInteger, SentryMemoryUnit) {
  SentryMemoryUnitByte,
  SentryMemoryUnitMegabyte,
...
}

typedef struct {
  int quantity;
  SentryMemoryUnit unit;
} SentryMemorySize;
```

maybe overkill, just looking at alternatives. I'm mostly fine with the proposed typedef. One benefit of having the enum/struct is that doc comments can be attached that can be brought up with a doc popover in Xcode with an ⌥ + click on the names.

**Pull request merged**